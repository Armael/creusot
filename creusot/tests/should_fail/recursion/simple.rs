#![allow(unused)]
extern crate creusot_contracts;
use creusot_contracts::*;

fn recurses(b: bool) {
    if b {
        recurses(!b);
    }
}
