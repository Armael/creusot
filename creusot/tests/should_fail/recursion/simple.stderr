error: Recursive function without a `#[variant]` clause
 --> simple.rs:5:1
  |
5 | fn recurses(b: bool) {
  | ^^^^^^^^^^^^^^^^^^^^
  |
note: Recursive call happens here
 --> simple.rs:7:9
  |
7 |         recurses(!b);
  |         ^^^^^^^^^^^^

error: aborting due to 1 previous error

